File: app/models/peca.py
from sqlmodel import SQLModel, Field
from datetime import date
import uuid
from uuid import UUID
from typing import Optional

class Peca(SQLModel, table=True):
    id: UUID = Field(
        default_factory=uuid.uuid4,
        primary_key=True,
        index=True,
        nullable=False
    )
    descricao: str
    status: str = Field(default="Pendente")
    valor: float = Field(default=0.0)
    data_entrega: date = Field(index=True)
    cliente_id: UUID = Field(foreign_key="cliente.id")

File: ui/front_end/lib/models/peca.dart
class Peca {
  final String id;
  final String descricao;
  final double valor;
  final String dataEntrega;
  final String clienteId;
  final String status;

  Peca({
    required this.id, required this.descricao, required this.valor,
    required this.dataEntrega, required this.clienteId, required this.status
  });

  factory Peca.fromJson(Map<String, dynamic> json) {
    return Peca(
      id: json['id'],
      descricao: json['descricao'],
      valor: (json['valor'] as num).toDouble(),
      dataEntrega: json['data_entrega'],
      clienteId: json['cliente_id'],
      status: json['status'],
    );
  }
}